zenohd_vendor/CMakeLists.txt

```diff
@@ -0,0 +1,33 @@
+cmake_minimum_required(VERSION 3.16)
+project(zenohd_vendor)
+
+# Default to C++17
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 17)
+  set(CMAKE_CXX_STANDARD_REQUIRED ON)
+endif()
+
+find_package(ament_cmake REQUIRED)
+
+include(ExternalProject)
+
+ExternalProject_Add(
+    zenoh
+    PREFIX ${CMAKE_BINARY_DIR}/_cargo_deps
+    GIT_REPOSITORY https://github.com/eclipse-zenoh/zenoh.git
+    GIT_TAG master
```

Comment:
I think for consistency, we should probably pin this to a particular hash.  But I think we should also do the same thing for zenoh_c_vendor, so I'll open a separate PR doing that.